File: Tests/ImportStandalone.m
#if WEBP_DECODER_ONLY
@import WebPDecoder;
#else
@import WebP;
@import WebP.Decoder;
#endif
int main(void) {
    const unsigned char malformed[] = {0, 1, 2, 3};
    int width, height;
    if (WebPDecodeRGBA(malformed, sizeof(malformed), &width, &height) != 0) return 1;
#if !WEBP_DECODER_ONLY
    if (WebPGetEncoderVersion() == 0) return 1;
#endif
    return WebPGetDecoderVersion() == 0;
}

File: Tests/ImportStandalone.swift
#if WEBP_DECODER_ONLY
import WebPDecoder
#else
import WebP
import WebP.Decoder
#endif

precondition(WebPGetDecoderVersion() > 0)
#if !WEBP_DECODER_ONLY
precondition(WebPGetEncoderVersion() > 0)
#endif
// Pull in the real decoder as well as the version function, without any other
// WebP module's headers or libraries being present on the search/link paths.
let malformed: [UInt8] = [0, 1, 2, 3]
malformed.withUnsafeBufferPointer { bytes in
    var width: Int32 = 0
    var height: Int32 = 0
    precondition(WebPDecodeRGBA(bytes.baseAddress, bytes.count, &width, &height) == nil)
}
print("Standalone Swift module import/link passed")
